lib/AST/ASTPrinter.cpp

```diff
@@ -276,6 +276,21 @@ PrintOptions PrintOptions::printSwiftInterfaceFile(ModuleDecl *ModuleToPrint,
           return false;
       }
 
+      // Skip enum cases containing enum elements we wouldn't print.
+      if (auto *ECD = dyn_cast<EnumCaseDecl>(D)) {
+        auto elements = ECD->getElements();
+        if (!elements.empty()) {
+          // Enum elements are usually not printed, so we have to override the
+          // print option controlling that. We only check the first element
+          // because all the elements in a single case decl should have the same
+          // characteristics.
```

Comment:
Should we assert that all the elements return the same `shouldPrint` result? otherwise, LGTM!

Reply:
I think that if were to do this kind of validation I would rather create some helper APIs on `EnumCaseDecl` to codify that using any element is appropriate for this kind of check and then assert during parsing that we parse all elements with the same attributes. This approach of grabbing the first case is already used in a couple of disparate locations so asserting this here seems insufficiently general.

Reply:
I've added a `getFirstElement()` utility and adopted it throughout to codify this practice.

Reply:
Thank you!